File: tp1/exemple/src/conv.py
#!/usr/local/bin/python3

import sys
import time

from utils import print_matrix, read_matrix

ex_path1 = sys.argv[1]  # Path de la première matrice
ex_path2 = sys.argv[2]  # Path de la deuxième matrice


def multiply_conv(mat_1, mat_2):
    length = len(mat_1)
    result = [0] * length
    result = list(map(lambda x: [0] * length, result))

    for i in range(length):
        for j in range(length):
            for k in range(length):
                result[i][j] += mat_1[i][k]*mat_2[k][j]

    return result


if __name__ == "__main__":
    a = read_matrix(ex_path1)
    b = read_matrix(ex_path2)

    start = time.time()
    matrix = multiply_conv(a, b)
    duration = (time.time() - start) * 1000

    options = sys.argv[3:]
    if '-p' in options:  # On imprime la matrice résultat
        # Données bidon, mais output du bon format demandé
        print_matrix(matrix)
    if '-t' in options:  # On imprime le temps d'exécution
        print(duration)  # Données bidon, mais output du bon format demandé ok

File: tp1/exemple/cpp_src/conv.cpp
#include <vector>
#include <ctime>
#include "multiplier.h"

using namespace std;

vector<int> multiplyConv(vector<int> &mat1, vector<int> &mat2)
{
    int len = mat1.size();
    vector<int> result(len);
    int rowSize = (int)sqrt(len);

    for (size_t i = 0; i < rowSize; i++)
    {
        for (size_t j = 0; j < rowSize; j++)
        {
            int curr = 0;
            for (size_t k = 0; k < rowSize; k++)
            {
                curr += mat1[i * rowSize + k] * mat2[k * rowSize + j];
            }
            result[i * rowSize + j] = curr;
        }
    }

    return result;
}